tests/Microsoft.Bot.Builder.Core.Extensions.Tests/CatchException_MiddlewareTests.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Bot.Builder.Core.Extensions.Tests
+{
+    [TestClass]
+    public class CatchException_MiddlewareTests
+    {
+        [TestMethod]
+        [TestCategory("Middleware")]
+        public async Task CatchException_TestMiddleware()
+        {
+            TestAdapter adapter = new TestAdapter()
+                .Use(new CatchExceptionMiddleware((context, exception) =>
+                {
+                    context.SendActivity(context.Request.CreateReply("Sorry, something went wrong"));
```

Comment:
Vette that the exception passed in is:
1. The correct type.
2. Has the right description ("test"). Otherwise weird stuff might be going on.

Reply:
Now that the middleware takes a specific generic type of exception and we are checking for the description as part of the tests I think we should be good here.